src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/BotSegments/KeyFrameEvaluator.cs

```diff
@@ -184,6 +182,13 @@ internal bool MatchedHelper(bool firstSegment, int segmentNumber, BooleanCriteri
                      case KeyFrameCriteriaType.CVImage:
                         cvImagesToMatch.Add(entry);
                         break;
+                    case KeyFrameCriteriaType.BehaviourComplete:
```

Comment:
Now that I see this all put together.. what if we made this `ActionComplete` , would that work / be more powerful so that we can just wait for any segment action to complete ?

Reply:
I'll +1 this idea of making it more generic so we can use it elsewhere!